jobs/MeterInstanceJob.js

```diff
@@ -0,0 +1,158 @@
+      }
+    };
+    return apiServerClient.getResources(options);
+  }
+
+  /* jshint ignore:start */
+  static async meter(events) {
+    try {
+      logger.info(`Number of events to be metered in this run - ${events.length}`);
+      // Adding this comment as we are transitioning to async/await
+      // Note: The below Promise is not bluebird promise
+      let promiseArray = await Promise.all(_.map(events, async (event) =>
+        await this.sendEvent(event)));
+      let successCount = promiseArray.filter(r => r === true).length;
+      return {
+        totalEvents: events.length,
+        success: successCount,
+        failed: events.length - successCount,
+      }
+    } catch (err) {
+      logger.error(`Error occurred while metering all events : `, err);
+    }
+  }
+  /* jshint ignore:end */
+
+  static isServicePlanExcluded(options) {
+    const serviceId = _.get(options, 'service.service_guid');
+    const planId = _.get(options, 'service.plan_guid');
+    logger.info(`Checking if service ${serviceId}, plan: ${planId} is excluded`);
+    const plan = catalog.getPlan(planId);
+    return !plan.metered;
+  }
+
+  static enrichEvent(options) {
+    // Add region , service name and plan sku name of the event
+    const serviceId = _.get(options, 'service.service_guid');
+    const planId = _.get(options, 'service.plan_guid');
+    logger.info(`Enriching the metering event ${serviceId}, plan: ${planId}`);
+    options.service.id = catalog.getServiceName(serviceId);
+    options.service.plan = catalog.getPlanSKUFromPlanGUID(serviceId, planId);
+    options.service = _.omit(options.service, ['service_guid', 'plan_guid']);
+    options.consumer.region = config.metering.region;
+    return options;
+  }
+
+  static getPlanSKUFromPlanGUID(serviceGuid, planGuid) {
+    logger.info(`Getting Plan SKU for service ${serviceGuid}, plan: ${planGuid}`);
+    const service = _.chain(catalog.toJSON().services)
+      .map((s) => s.id === serviceGuid ? s : undefined)
+      .filter(s => s !== undefined)
+      .head()
+      .value();
+    return _
+      .chain(service.plans)
+      .map((p) => p.id === planGuid ? p.sku_name : undefined)
+      .filter(p => p !== undefined)
+      .head()
+      .value();
+  }
+
+  /* jshint ignore:start */
+  static async sendEvent(event) {
+    try {
+      logger.debug('Metering Event details before enriching:', event);
+      if (this.isServicePlanExcluded(event.spec.options) === true) {
+        await this.updateMeterState(CONST.METER_STATE.EXCLUDED, event);
+        return true;
+      }
+      let enrichedUsageDoc = await this.enrichEvent(_.get(event.spec,'options'));
+      logger.info('Sending enriched document:', enrichedUsageDoc);
+      let validEvent = await maas.client.sendUsageRecord({
+        usage: [enrichedUsageDoc]
+      });
+      validEvent ? this.updateMeterState(CONST.METER_STATE.METERED, event) : false;
```

Comment:
I feel it would be better to use standard `if-else` instead of ternary here for better readability